bundles/mapping/mapmodule/oskariStyle/generator.ol.js

```diff
@@ -1,15 +1,223 @@
-
 import olStyleStyle from 'ol/style/Style';
 import olStyleFill from 'ol/style/Fill';
 import olStyleStroke from 'ol/style/Stroke';
 import olStyleCircle from 'ol/style/Circle';
 import olStyleIcon from 'ol/style/Icon';
 import olStyleText from 'ol/style/Text';
 
-import { LINE_DASH, LINE_JOIN, EFFECT, FILL_STYLE } from './constants';
+import { LINE_DASH, LINE_JOIN, EFFECT, FILL_STYLE, STYLE_TYPE } from './constants';
+import { filterOptionalStyle, getOptionalStyleFilter } from './filter';
 
+const log = Oskari.log('MapModule.util.style');
 const TRANSPARENT = 'rgba(0,0,0,0)';
 
+const merge = (...styles) => jQuery.extend(true, {}, ...styles);
+
+export const HIDDEN_STYLE = new olStyleStyle({ visibility: 'hidden' });
+
+const defaultStyles = {};
+export const setDefaultStyle = (layerType, styleDef) => {
+    defaultStyles[layerType] = styleDef;
+};
+const getDefaultStyle = layerType => {
+    return defaultStyles[layerType] || {};
+};
+
+const getFeatureStyle = (layer, extendedDef = {}) => {
+    const style = layer.getCurrentStyle();
+    const defaultDef = getDefaultStyle(layer.getLayerType());
+    return merge(defaultDef, style.getFeatureStyle(), extendedDef);
+};
+
+export const useStyleFunction = layer => {
```

Comment:
I'm not sure what would be a good name for this but useX() signals that some operation is performed for the current layer instead of the parameter layer is used to detect if the layer requires a function to be used for styling. 
```suggestion
export const doesLayerRequireStyleFunction = layer => {
```
I haven't looked where this is used but could the "function vs some other styling value" be detected without this?